### PR TITLE
fix(catalogue): dcat semantics

### PR DESCRIPTION
**Pull request by BrendaHijmans**
### What are the main changes you did
- some changes in semantic annotations, to comply to HRICore 1.0 (and DCAT-AP3.0/NL) and towards HRICore 2.0 and HealthDCAT-AP.
- see diff for changes

>dct:issued and dct:modified are now generated from mg_insertedOn and mg_updatedOn. Upon new import of data (e.g. for migrations) these columns are set to the date of the new import.
>need: pids (present in umcg research data catalogue)
>need: RORs

>add dcat:theme = Health (<skos:prefLabel xml:lang="en">Health</skos:prefLabel> / http://publications.europa.eu/resource/authority/data-theme/HEAL)
>can dct:creator be dependent on Organisations.is lead organisation = true (should actually be Organisations.role = Data originator, but this is seldom filled out) > use compute field?
>can dct:publisher be dependent on Organisations.is lead organisation = true > use compute field?

>dct:license for dcat:Dataset is not filled out by any resource (Resources.license), can we use a place holder? HealthDCAT-AP is aiming for a mandatory url that at least needs to be filled out here and extra can be added as well.
>same for dct:publisher for dcat:Catalog

>if a Network is a Catalogue: get dcat:dataset from Resources table.

### How to test
- explain here what to do to test this (or point to unit tests)

### Checklist
- [ ] make ref naar Contact die als contacpoint moet worden opgegeven?
- [ ] dcterms creator en publisher aparte velden die verwijzen naar Organisations

- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

File: data/_models/shared/Catalogues.csv [hunk 1]
```diff
@@ -1,7 +1,7 @@
 tableName,tableExtends,columnName,columnType,key,required,refSchema,refTable,refLink,refBack,validation,semantics,description,profiles,visible,computed
 Catalogues,,,,,,,,,,,dcat:Catalog,A collection of resources within a network or consortium or about a common topic,DataCatalogueFlat,,
-Catalogues,,name,,1,true,,,,,,dct:title,The name given to the catalogue,DataCatalogueFlat,,
-Catalogues,,network,ref,1,true,,Resources,,,,,Network or consortium that publishes this catalogue,DataCatalogueFlat,,
-Catalogues,,type,ontology,,true,CatalogueOntologies,CatalogueTypes,,,,dcat:theme,Type of catalogue,DataCatalogueFlat,,
```
Comment (mswertz) [inline data/_models/shared/Catalogues.csv:5]: waarom theme eraf gehaald?

Comment (mswertz) [reply data/_models/shared/Catalogues.csv:5]: of vind je dit geen 'theme'. Let wel, RDF heeft er geen enkel probleem mee als wij meerdere velden naar hetzelfde dcat item mappen. Die theme tags komen voor de buitenwereld gewoon in 1 lijstje.

File: data/_models/shared/Organisations.csv [hunk 2]
```diff
@@ -1,12 +1,12 @@
 tableName,tableExtends,columnName,columnType,key,required,refSchema,refTable,refLink,refBack,validation,semantics,description,profiles,visible
-Organisations,,,,,,,,,,,"org:Organization,foaf:Agent",Research departments and research groups. Null values will be generated from RoR,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
-Organisations,,resource,ref,1,TRUE,,Resources,,,,dcterms:isPartOf,Resource that this organisation is part of,"DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
+Organisations,,,,,,,,,,,foaf:Agent,Research departments and research groups. Null values will be generated from RoR,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
```
Comment (mswertz) [inline data/_models/shared/Organisations.csv:2]: waarom org:Organization eraf gehaald? Ik dacht dat dat dcat recommendation was?

File: data/_models/shared/Organisations.csv [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+Organisations,,resource,ref,1,TRUE,,Resources,,,,,Resource that this organisation is part of,"DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
 Organisations,,id,,1,TRUE,,,,,,dcterms:identifier,Internal identifier,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
 Organisations,,pid,hyperlink,,,,,,,,dcterms:identifier,"Persistent identifier, ideally from ROR","DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
-Organisations,,name,text,,TRUE,,,,,,"dcterms:title,foaf:name",Name used in (inter)national projects,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
-Organisations,,acronym,,,,,,,,,dcterms:identifier,"Acronym, if applicable","DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
```
Comment (mswertz) [inline data/_models/shared/Organisations.csv:7]: dit ook, waarom dit allemaal weghalen, heb je er last van?

File: data/_models/shared/Organisations.csv [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+Organisations,,name,text,,TRUE,,,,,,foaf:name,Name used in (inter)national projects,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
+Organisations,,acronym,,,,,,,,,,"Acronym, if applicable","DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
 Organisations,,logo,file,,,,,,,,,Logo of the organisation,"DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging",
-Organisations,,country,ontology_array,,,CatalogueOntologies,Countries,,,,vcard:country-name,Country in which the institution head office or coordinating centre is located,"DataCatalogueFlat,EMA,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging,UMCUCohorts",
```
Comment (mswertz) [inline data/_models/shared/Organisations.csv:9]: ook hier is niets mis mee volgens mij?

File: data/_models/shared/Resources.csv [hunk 5]
```diff
@@ -20,7 +20,7 @@ Resources,,external identifiers,refback,,,,External identifiers,,resource,,,Exte
 Resources,,start year,int,,,,,,,,dcat:startDate,"The year when the resource started, e.g. when data or samples started to be collected or extracted, or when the resource was established","DataCatalogueFlat,EMA,RWEStaging,CohortsStaging,UMCGCohortsStaging,NetworksStaging"
 Resources,,end year,int,,,,,,,,dcat:endDate,"The year the resource ceased, e.g. when data or sample collection was stopped","DataCatalogueFlat,EMA,RWEStaging,CohortsStaging,UMCGCohortsStaging,NetworksStaging"
 Resources,,time span description,text,,,,,,,,,Description of time span,"DataCatalogueFlat,RWEStaging"
-Resources,,contact email,,,,,,,,,dcat:contactPoint,Contact e-mail address ,"DataCatalogueFlat,CohortsStaging,RWEStaging,NetworksStaging,StudiesStaging,UMCGCohortsStaging"
```
Comment (mswertz) [inline data/_models/shared/Resources.csv:23]: dito

File: data/_models/shared/Resources.csv [hunk 6]
```diff
@@ -35,8 +35,8 @@ Resources,,record trigger,text,,,,,,,,,"What triggers the creation of a record i
 Resources,,unit of observation,text,,,,,,,,,"Based on the record trigger or promt, what is the unit(s) of observation of a record (e.g. person, prescription)?","DataCatalogueFlat,RWEStaging"
 Resources,,subpopulations,refback,,,,Subpopulations,,resource,,,"List of cohorts or subpopulations defined in this resource, if applicable","DataCatalogueFlat,CohortsStaging,UMCGCohortsStaging,NetworksStaging,UMCUCohorts"
 Resources,,collection events,refback,,,,Collection events,,resource,,dcterms:hasPart,"List of data/sample collection events defined for this collection, if applicable","DataCatalogueFlat,CohortsStaging,UMCGCohortsStaging,NetworksStaging,UMCUCohorts"
-Resources,,resources,ref_array,,,,Resources,,,,dct:dataset,List of other resources included in this resource,"DataCatalogueFlat,EMA"
-Resources,,part of resources,refback,,,,Resources,,resources,,dct:dataset,List of other collections included in this resource,"DataCatalogueFlat,EMA"
+Resources,,resources,ref_array,,,,Resources,,,,dcat:dataset,List of other resources included in this resource,"DataCatalogueFlat,EMA"
```
Comment (mswertz) [inline data/_models/shared/Resources.csv:38]: heb je gecheckt welke prefix we nu gebruiken op RDF api nivo?

**Review by mswertz — CHANGES_REQUESTED** [review]
ff bespreken: je hebt allerlei dingen weggehaald, die volgens mij gewoon goed zijn, waarom?

Enige wat nu nodig is dat we checken dat we WEL hebben wat er nodig is. Zoals die missende dct:publisher/creator.

Het is normaal niet problematisch als we meer hebben tenzij conflicterend met DCAT-health. Dwz, als de validators erover vallen.

File: data/_models/shared/Catalogues.csv (hunk 1 above)
Comment (BrendaHijmans) [reply data/_models/shared/Catalogues.csv:5]: dcat:Catalog class does not have a dcat:theme, but dcat:themeTaxonomy. And this is a controlled vocabulary. So we should just write it in the rdf, I guess. https://healthdcat-ap.github.io/#Catalogue.themes says: "This property refers to a knowledge organisation system used to classify the Catalogue's Datasets. It must have at least the value NAL:data-theme as this is the mandatory controlled vocabulary for dcat:theme."

File: data/_models/shared/Organisations.csv (hunk 2 above)
Comment (BrendaHijmans) [reply data/_models/shared/Organisations.csv:2]: ik zet die annotatie weer terug.

File: data/_models/shared/Organisations.csv (hunk 3 above)
Comment (BrendaHijmans) [reply data/_models/shared/Organisations.csv:7]: dct:identifier is not used within foaf:Agent or org:Organization class and acronym is not an identifier, the pid/ror is. changed to skos:altLabel.

dcterms:title is not used within foaf:Agent or org:Organization class. changed to skos:prefLabel

File: data/_models/shared/Organisations.csv (hunk 4 above)
Comment (BrendaHijmans) [reply data/_models/shared/Organisations.csv:9]: I am not sure, see this about locations in Organization class: https://www.w3.org/TR/vocab-org/#location_information. Should we annotate this attribute with org:hasSite and the Countries table with org:Site?

File: data/_models/shared/Resources.csv (hunk 5 above)
Comment (BrendaHijmans) [reply data/_models/shared/Resources.csv:23]: dcat:contactPoint is structured in vcard:Kind, which in HRICore and DCAT-AP3.0 needs a name and email address. So I removed this annotation, but kept the one on personsInvolved.

File: data/_models/shared/Resources.csv (hunk 6 above)
Comment (BrendaHijmans) [reply data/_models/shared/Resources.csv:38]: dct and dcat are different things, dct is dublin core, dcat is dcat. for dct the prefix in the rdf is dcterm, so I will also check that.

File: data/_models/shared/Catalogues.csv (hunk 1 above)
Comment (mswertz) [reply data/_models/shared/Catalogues.csv:5]: And you are right we could have this 'NAL:data-theme' hardcoded?

File: data/_models/shared/Organisations.csv (hunk 4 above)
Comment (mswertz) [reply data/_models/shared/Organisations.csv:9]: ah, probably need sem experts opinion @esthervanenckevort ?

File: data/_models/shared/Resources.csv (hunk 5 above)
Comment (mswertz) [reply data/_models/shared/Resources.csv:23]: so we want to drop 'contact email' and instead use the contacts?

File: data/_models/shared/Organisations.csv (hunk 4 above)
Comment (esthervanenckevort) [reply data/_models/shared/Organisations.csv:9]: The countries table has countries in it, which are geographical regions. A site in the context of an organisation is something different, a site has usually has a siteAddress.
Semantically you could model it as follows

`
<https://www.umcg.nl/> a org:Organization
   hasSite _addr
      siteAddress: [ 
       vcard:hasAddress [
           vcard:country-name: "NL" 
        ]
      ]
    ] .
`

You could also see the Organisation as an instance of a vCard (next to Organization) with vCard type "Organization" and use the hasAddress property directly on the RDF.

`
<https://www.umcg.nl/> a org:Organization, vcard:Organization
   vcard:hasAddress [
       vcard:country-name: "NL" 
   ] .
`

However in both cases the country cannot be a direct property of Organisation.

If you would like to have country at the level of Organisation you should define your own property that has a org:Organization as domain.

I inlined blank nodes for brevity.

Comment (BrendaHijmans) [reply data/_models/shared/Organisations.csv:9]: I am not sure what to do here still. Better not to include country in the rdf output if it is not asked for anywhere? Or do we want to have it at the level of Organization for other purposes, maybe for ingesting data? And how do we define our "own property that has a org:Organization as domain"?

**Review by mswertz — APPROVED** [review]
need to fix merge conflict but given our joint review last time that can be merged.

**Pull request merged**